apps/lens_blur/process.cpp

```diff
@@ -21,7 +21,10 @@ int main(int argc, char **argv) {
     // Let the Halide runtime hold onto GPU allocations for
     // intermediates and reuse them instead of eagerly freeing
     // them. cuMemAlloc/cuMemFree is slower than the algorithm!
-    halide_reuse_device_allocations(nullptr, true);
```

Comment:
Not currently, no, but since the calling contact allows for an error to be returned, and the hook is user-replaceable, we should honor it (or change the contract)

Reply:
Let's instead change the comment on this function in HalideRuntime.h to explicitly say that if the flag is set to true, the call will always succeed. There are multiple reasons why this can never return non-zero, but that's the big one.

Reply:
If we don't think it should ever return nonzero, let's just document that, period -- i.e., "For historical reasons, this call returns an error code, but it will always be halide_error_code_success; if you reimplement this call in the runtime, you must honor this contract."

Reply:
It's only guaranteed to succeed when the second arg is set to true.

Reply:
I really don't think we should change this code. It's a demo app, which means we should be brutal about eliminating anything not strictly necessary. I don't want readers pausing to take in the fact that this can't throw an error.

Reply:
Ignoring the error code from a function that returns an error code is bad practice. I can live with removing the comment, but the `(void)` should remain.

Reply:
> It's a demo app, which means we should be brutal about eliminating anything not strictly necessary.

Let me be a bit more detailed in my concern: a demo app needs to strike a balance between avoiding stuff that is unnecessary (in the service of making the point), but leaving in details that really are necessary in a real-world app. My concern is that if we leave error-checking out of demo apps, people will cut-n-paste this code as examples, and end up using code that isn't robust enough for real-world apps.

In the specific case of `halide_reuse_device_allocations()` (and the underlying `release_unused()` hook function), I think maybe we should just remove the error code entirely -- it's not clear to me how any implementation we replaced it with would likely return an error in the first place.

Reply:
Open question: is there a convention we can use in apps (etc) that satisfies the following:

- Unambiguously indicates that a function returns an error that needs checking in production code
- Unambiguously indicates that we are deliberately ignoring possible errors in this case for the sake of brevity
- Is unlikely to get copy-pasted and become Default Best Practice via developer laziness

?